### PR TITLE
.Net: Fix bug to allow activation of JSON mode through execution_settings

### PR DESCRIPTION
**Pull request by sugii-norichika-intern-fixer**
### Motivation and Context
Closes [#4719](https://github.com/microsoft/semantic-kernel/issues/4719)

**Issue**

- When setting the `execution_settings` from YAML or `config.txt`, the `response_format` field is not recognized, and thus the JSON mode cannot be activated.

**Root Cause**
- The issue arises because the `ResponseFormat` property, which is of type `object?`, stores values as `JsonElement` types when read from the aforementioned files.
- The `CreateChatCompletionOptions` method does not account for this and performs type checking directly against types such as `string`, leading to improper conditional branching.
- As a result, the `ChatCompletionsOptions`'s `ResponseFormat` value remains unassigned, preventing the activation of JSON mode.

### Description

This PR addresses the problem by adjusting the type checks to properly recognize the `JsonElement` objects.
 This ensures that when `ResponseFormat` is set, the request sent to OpenAI is correct, and JSON mode can be utilized.

Please let me know if any further changes or discussions are needed regarding this pull request.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by sugii-norichika-intern-fixer** [comment]
@microsoft-github-policy-service agree company="FIXER Inc."

File: dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs [hunk 1]
```diff
@@ -781,6 +781,28 @@ private static CompletionsOptions CreateCompletionsOptions(string text, OpenAIPr
                         break;
                 }
                 break;
+
+            case JsonElement formatElement:
+                // This is a workaround for a type mismatch when deserializing a JSON into an object? type property.
+                // Handling only string formatElement.
+                if (formatElement.ValueKind == JsonValueKind.String)
+                {
+                    string? formatString = formatElement.GetString();
```
Comment (ebihara-kohta23-fixer) [inline dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs:790]:
```suggestion
                    string formatString = formatElement.GetString() ?? "";
```
I think this is slightly better than using if statements for null checks.

**Review by ebihara-kohta23-fixer — COMMENTED** [review]
Commented some suggestions

**Comment by sugii-norichika-intern-fixer** [comment]
@markwallace-microsoft 
 Hi there, I've made the necessary code adjustments to pass CI. It would be great if you could take a look. Thanks!

**Review by markwallace-microsoft — APPROVED** [review]

**Review by SergeyMenshykh — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**